type_lens/function_view.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import inspect
+import sys
+from typing import TYPE_CHECKING, Any, Callable
+
+from typing_extensions import get_type_hints
+
+from type_lens.parameter_view import ParameterView
+from type_lens.type_view import TypeView
+
+__all__ = ("FunctionView",)
+
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+
+class FunctionView:
+    def __init__(self, fn: Callable, type_hints: dict[str, type]):
+        self.function = fn
+        self.signature = getattr(fn, "__signature__", None) or inspect.signature(fn)
+
+        self.return_type = TypeView(type_hints.pop("return", None))
+
+        self.parameters = tuple(
+            ParameterView.from_parameter(param, type_hints) for param in self.signature.parameters.values()
+        )
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, FunctionView):
+            return False
+
+        return bool(
+            self.function == other.function
+            and self.parameters == other.parameters
+            and self.return_type == other.return_type
+        )
+
+    def __repr__(self) -> str:
+        cls_name = self.__class__.__name__
+
+        return f"{cls_name}({self.function.__name__})"
+
+    @classmethod
+    def from_type_hints(cls: type[Self], fn: Callable, include_extras: bool = False) -> Self:
```

Comment:
Would it make more sense for this to be `from_callable()`? IMO `from_type_hints()` implies that the arg should be the type hints.

Reply:
mmmm yes. well `from_<function or callable, depending on the name of the class>`